### PR TITLE
Identifier & footer external link wrapping in safari

### LINKED ISSUE #1928: USWDS-Site - Bug: Remove extra whitespace in identifier links
### Describe the bug

There are whitespaces being rendered before/after links in identifier. This is causing the external link to break into it's own line in Safari. 

**Screenshots from uswds/uswds#5046**
image

image

### Steps to reproduce the bug

1. Open Safari
2. Go to https://designsystem.digital.gov/
3. Inspect identifier required links
4. Verify white spaces are before/after string

### Expected Behavior

There should be no whitespace in rendered text.

### Screenshots

_No response_

### System setup

- Safari Version 16.1 (18614.2.9.1.12)

### Additional context

Related to https://github.com/uswds/uswds/pull/5046.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate bug reports.

### LINKED ISSUE #1953: USWDS-Site - Feature: Templatize footer
### Is your feature request related to a problem? Please describe.

Templatizing footer will remove extra whitespace being added in Safari. Related to #1928.

### Describe the solution you'd like

- [ ] Create a data file for footer content
- [ ] Update template to use this data

### Describe alternatives you've considered

n/a

### Additional context

https://github.com/uswds/uswds-site/pull/1918#pullrequestreview-1202890602

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

### PR DESCRIPTION
**Pull request by mejiaj**
## Description

- **Site identifier has been converted into a template.**  Removes whitespace issues in Safari where spaces were being rendered before/after string. Closes #1928. [[dbb6795]](https://github.com/uswds/uswds-site/pull/1918/commits/dbb6795c41e81080fee8d7a9ecbada8819046306)
- **Site footer has been converted into a template.** Closes #1953. [[a103d4c](https://github.com/uswds/uswds-site/pull/1918/commits/a103d4c0c6036790790669cb284e88756a89c63e)
- Test branch for https://github.com/uswds/uswds/pull/5046.

~⚠️ uswds/uswds#5046 must be merged **before** this PR.~

Small part relies on uswds/uswds#5046 for the whitespace issue on safari, but has improvements that can be merged. 

## Content editors

**Content for site Identifier and Footer is now done in YML files.** Instead of updating template markup you can go to `_data/footer.yml` and `_data/identifier.yml` to update content.

## Preview links
- [Homepage identifier](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-site-external-link-wrapping/)
- [Link component](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-site-external-link-wrapping/components/link/)

## Tested

I tested the following areas shown below.

**Component preview**
image

**External link guidance section**
image

---

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

File: _includes/identifier.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          {{- identifier.disclaimer -}}
+        </p>
+      </section>
+    </div>
+  </section>
+  <nav class="usa-identifier__section usa-identifier__section--required-links" aria-label="Important links">
+      <div class="usa-identifier__container">
+        <ul class="usa-identifier__required-links-list">
+          {%- for link in identifier.requiredLinks -%}
+          <li>
```
Comment (amyleadem) [inline _includes/identifier.html:35]: I noticed that there was a visual difference in the identifier at mobile widths. Looks like the `<li>` elements might be missing `.usa-identifier__required-links-item`.

image

**Review by amyleadem — CHANGES_REQUESTED** [review]
Identifier is looking good. I noticed one missing class ([details](https://github.com/uswds/uswds-site/pull/1918#discussion_r1038396145)), but the line break issue seems to be resolved.

I also noticed that the issue still appears on the external links in the footer:
image

Can you take a look at that as well?

**Comment by mejiaj** [comment]
> Identifier is looking good. I noticed one missing class ([details](https://github.com/uswds/uswds-site/pull/1918#discussion_r1038396145)), but the line break issue seems to be resolved.
> 
> I also noticed that the issue still appears on the external links in the footer: image
> 
> Can you take a look at that as well?

@amyleadem fixed in [a103d4c](https://github.com/uswds/uswds-site/pull/1918/commits/a103d4c0c6036790790669cb284e88756a89c63e).

**Safari Version 16.1 (18614.2.9.1.12)**
image

File: _includes/footer.html [hunk 2]
```diff
@@ -1,154 +1,47 @@
...
+          <div class="tablet:grid-col">
+            <div class="usa-media-block">
+              <div class="usa-media-block__img circle-5 bg-accent-cool display-flex flex-row flex-align-center flex-justify-center text-white">
+                <svg class="usa-icon usa-icon--size-3" aria-hidden="true" role="img" focusable="false">
+                  <use xlink:href="{{ site.baseurl }}/assets/img/sprite.svg#{{ callout.icon }}"></use>
+                </svg>
+              </div>
+              <div class="usa-media-block__body">
+                <h3>{{ callout.tagline }}</h3>
+                <a href="{{ callout.link_href }}">
```
Comment (amyleadem) [inline _includes/footer.html:25]: Should these have at least the `usa-link` class?

**Review by amyleadem — CHANGES_REQUESTED** [review]
@mejiaj The footer is looking good! I had one [question](https://github.com/uswds/uswds-site/pull/1918/files#r1054583440) about the `usa-link` class. I also noticed that the [identifier list items](https://github.com/uswds/uswds-site/pull/1918/files#r1038396145) are still missing the `usa-identifier__required-links-item` class. Can you take a look at those?

File: _includes/footer.html (hunk 2 above)
Comment (mejiaj) [reply _includes/footer.html:25]: @amyleadem normally I would say yes, but other site links are styled globally. I didn't want to go against that pattern.

**Comment by mejiaj** [comment]
> 



> @mejiaj The footer is looking good! I had one [question](https://github.com/uswds/uswds-site/pull/1918/files#r1054583440) about the `usa-link` class. I also noticed that the [identifier list items](https://github.com/uswds/uswds-site/pull/1918/files#r1038396145) are still missing the `usa-identifier__required-links-item` class. Can you take a look at those?

Added missing class in e8fa6f7b6ccfe1ac07842653b727b723a673bc95. Good catch!

File: _data/identifier.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  - label: No FEAR Act data
+    href: https://www.gsa.gov/reference/civil-rights-programs/notification-and-federal-employee-antidiscrimination-and-retaliation-act-of-2002
+  - label: Office of the Inspector General
+    href: https://www.gsaig.gov/
+  - label: Performance reports
+    href: https://www.gsa.gov/reference/reports/budget-performance
+  - label: Privacy policy
+    href: https://www.gsa.gov/website-information/website-policies
+usagov:
+  description: Looking for U.S. government information and services
```
Comment (amyleadem) [inline _data/identifier.yml:24]: Missing question mark
```suggestion
  description: Looking for U.S. government information and services?
```

**Review by amyleadem — APPROVED** [review]
Everything looks good. Thanks for cleaning this up!

FYI - I committed one small change to add a missing question mark in  [4c2b899](https://github.com/uswds/uswds-site/pull/1918/commits/4c2b8991d3dc9b28a5e85b04ec4423ba2c2f50ab)

:warning: Also, a reminder to update uswds in the package.json before merging

**Comment by mejiaj** [comment]
@amyleadem thanks for that and the great review, much appreciated! I've also set the branch to use the latest USWDS.

**Review by thisisdano — APPROVED** [review]
Nice collaboration getting this done!

**Pull request merged**